gef.py

```diff
@@ -7354,10 +7353,10 @@ def context_title(self, m: Optional[str]) -> None:
 
     def context_regs(self) -> None:
         self.context_title("registers")
-        ignored_registers = set(self["ignore_registers"].split())
```

Comment:
this can still be a set

Reply:
What value is there in converting to a set from what split returns?

Reply:
`x not in ignored_registers` is faster if ignored_registers is a set.

Reply:
I assume that there is some overhead to turning it into a set in the first place as it will need to hash all the entries. Is the saving from then comparing against a set significant enough that it's worth doing so?

Reply:
In order to compare against a set, the value being compared will either need to be directly compared by value (at which point the set is needless overhead) or the value will need to be hashed, compared by hash, and then compared by value if the hashes match (unless the hashes are guaranteed to not collide, which seems unlikely). I'm currently struggling to see a scenario where making this a `set` does anything but increase processing overhead.

Reply:
Because you use `not in ignored_registers` you not only need to compare (or hash) but you also need to iterate the whole list and compare EACH, unless you use a set.

If you `[reg for reg in gef.arch.all_registers if reg not in ignored_registers]` and `all_registers` is 20 elements, but `ignored_registers` is, say, 1 billion then you have to do 20*1billion/2 comparisons with a list, and 20*1 with a set. Sure, this is contrived, but it's still O(N*M) vs O(N)